src/commands/upload_file.ts

```diff
@@ -169,66 +169,29 @@ new CLICommand({
 				shouldBundle
 			});
 
-			const results = await Promise.all(
-				filesToUpload.map(async (fileToUpload) => {
-					const {
-						parentFolderId,
-						wrappedEntity,
-						destinationFileName,
-						drivePassword,
-						driveKey: fileDriveKey
-					} = fileToUpload;
-
-					return await (async () => {
-						if (await parameters.getIsPrivate()) {
-							const driveId = await arDrive.getDriveIdForFolderId(parentFolderId);
-							const driveKey =
-								fileDriveKey ??
-								(await parameters.getDriveKey({ driveId, drivePassword, useCache: true }));
-
-							if (isFolder(wrappedEntity)) {
-								return arDrive.createPrivateFolderAndUploadChildren({
-									parentFolderId,
-									wrappedFolder: wrappedEntity,
-									driveKey,
-									destParentFolderName: destinationFileName,
-									conflictResolution,
-									prompts: folderUploadConflictPrompts
-								});
-							} else {
-								return arDrive.uploadPrivateFile({
-									parentFolderId,
-									wrappedFile: wrappedEntity,
-									driveKey,
-									destinationFileName,
-									conflictResolution,
-									prompts: fileUploadConflictPrompts
-								});
-							}
-						} else {
-							if (isFolder(wrappedEntity)) {
-								return arDrive.createPublicFolderAndUploadChildren({
-									parentFolderId,
-									wrappedFolder: wrappedEntity,
-									destParentFolderName: destinationFileName,
-									conflictResolution,
-									prompts: folderUploadConflictPrompts
-								});
-							} else {
-								return arDrive.uploadPublicFile({
-									parentFolderId,
-									wrappedFile: wrappedEntity,
-									destinationFileName,
-									conflictResolution,
-									prompts: fileUploadConflictPrompts
-								});
-							}
-						}
-					})();
-				})
+			const uploadStats: ArDriveUploadStats[] = await Promise.all(
+				filesToUpload.map(
+					async ({ parentFolderId, wrappedEntity, destinationFileName, driveKey, drivePassword }) => {
+						driveKey ??= (await parameters.getIsPrivate())
```

Comment:
I assume that this operator is conditionally setting the value only if null. There are a couple of places in the codebase where this would spare us an `if` and thus one indentation level. Nice approach!

Reply:
Yes, that is the case: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_nullish_assignment